### PR TITLE
Finish edit profile

### PR DESCRIPTION
**Pull request by manhnhhedspi**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -28,8 +31,27 @@ def show
     end
   end
 
+  def edit
+    @user = User.find params[:id]
+  end
+
+  def update
+    @user = User.find params[:id]
```
Comment (phuongbk) [inline app/controllers/users_controller.rb]: make method set_user call before_action

**Comment by vuducdat** [comment]
ok

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -20,16 +25,37 @@ def create
   end
 
   def show
-    if User.exists? params[:id]
-      @user = User.find params[:id]
-    else
-      flash[:danger] = t "controllers.users.show.flash_danger"
+    if @user.nil?
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: one line of code

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,9 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:edit, :update]
+  before_action :set_user, only: [:edit, :update, :show]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: don't use `set` or `get` to set function name

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -20,16 +25,37 @@ def create
...
+
+  def correct_user
+    redirect_to root_url unless current_user? @user
+  end
+
+  def set_user
+    if User.exists? params[:id]
+      @user = User.find params[:id]
+    else
+      flash[:danger] = t "controllers.users.show.flash_danger"
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,4 +1,9 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:edit, :update]
+  before_action :load_user, only: [:edit, :update, :show]
+  before_action :correct_user, only: [:edit, :update]
+
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove one line

**Pull request merged**